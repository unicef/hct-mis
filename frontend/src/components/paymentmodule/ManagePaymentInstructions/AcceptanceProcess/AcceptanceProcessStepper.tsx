import Step from '@material-ui/core/Step';
import StepLabel from '@material-ui/core/StepLabel';
import Stepper from '@material-ui/core/Stepper';
import CancelIcon from '@material-ui/icons/Cancel';
import React from 'react';
import { useTranslation } from 'react-i18next';
import styled from 'styled-components';

const StyledCancelIcon = styled(CancelIcon)`
  color: #e90202;
`;

interface AcceptanceProcessStepperProps {
  acceptanceProcess;
}

export const AcceptanceProcessStepper = ({
  acceptanceProcess,
}: AcceptanceProcessStepperProps): React.ReactElement => {
  const { t } = useTranslation();
  const {
    rejectedOn,
    actions,
    authorizationNumberRequired,
    releaseNumberRequired,
  } = acceptanceProcess;
  const steps = [
    {
      name: `${t('Authorization')} (${
        actions.authorization.length
      }/${authorizationNumberRequired})`,
      hasError: rejectedOn === 'IN_AUTHORIZATION',
      isCompleted: actions.authorization.length === authorizationNumberRequired,
    },
    {
      name: `${t('Release')} (${
        actions.release.length
      }/${releaseNumberRequired})`,
      hasError: rejectedOn === 'IN_REVIEW',
      isCompleted: actions.release.length === releaseNumberRequired,
    },
  ];
  const getActiveStep = (): number => {
    if (actions.release.length === releaseNumberRequired) {
      return 1;
    }
    return 0;
  };

  return (
    <Stepper activeStep={getActiveStep()}>
      {steps.map((step) => (
        <Step completed={step.isCompleted} key={step.name}>
          <StepLabel
            error={step.hasError}
            StepIconComponent={step.hasError ? StyledCancelIcon : null}
          >
            {step.name}
          </StepLabel>
        </Step>
      ))}
    </Stepper>
  );
};
