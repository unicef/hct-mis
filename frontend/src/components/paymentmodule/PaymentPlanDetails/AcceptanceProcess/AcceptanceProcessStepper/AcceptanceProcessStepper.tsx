import { Box } from '@material-ui/core';
import Step from '@material-ui/core/Step';
import StepLabel from '@material-ui/core/StepLabel';
import Stepper from '@material-ui/core/Stepper';
import CancelIcon from '@material-ui/icons/Cancel';
import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import styled from 'styled-components';
import { PaymentPlanQuery } from '../../../../../__generated__/graphql';

const StyledBox = styled(Box)`
  width: 100%;
`;

const StyledCancelIcon = styled(CancelIcon)`
  color: #e90202;
`;

interface AcceptanceProcessStepperProps {
  acceptanceProcess: PaymentPlanQuery['paymentPlan']['approvalProcess']['edges'][0]['node'];
  approvalNumberRequired: number;
  authorizationNumberRequired: number;
  financeReviewNumberRequired: number;
}

export const AcceptanceProcessStepper = ({
  acceptanceProcess,
  approvalNumberRequired,
  authorizationNumberRequired,
  financeReviewNumberRequired,
}: AcceptanceProcessStepperProps): React.ReactElement => {
  const {
    sentForApprovalDate,
    sentForAuthorizationDate,
    sentForFinanceReviewDate,
    rejectedOn,
    actions,
  } = acceptanceProcess;

  const { t } = useTranslation();
  const getActiveStep = (): number => {
    if (sentForFinanceReviewDate) {
      return 2;
    }
    if (sentForAuthorizationDate) {
      return 1;
    }

    if (sentForApprovalDate) {
      return 0;
    }
    return 0;
  };

  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  const [activeStep, setActiveStep] = useState(() => getActiveStep());

  const steps = [
    {
      name: `${t('Approval')} (${actions.approval.length}/${approvalNumberRequired})`,
      hasError: rejectedOn === 'IN_APPROVAL',
      isCompleted: actions.approval.length === approvalNumberRequired,
    },
    {
      name: `${t('Authorization')} (${actions.authorization.length}/${authorizationNumberRequired})`,
      hasError: rejectedOn === 'IN_AUTHORIZATION',
      isCompleted: actions.authorization.length === authorizationNumberRequired,
    },
    {
      name: `${t('Finance Review')} (${actions.financeReview.length}/${financeReviewNumberRequired})`,
      hasError: rejectedOn === 'IN_REVIEW',
      isCompleted: actions.financeReview.length === financeReviewNumberRequired,
    },
  ];

  return (
    <StyledBox>
      <Stepper activeStep={activeStep}>
        {steps.map((step) => (
          <Step completed={step.isCompleted} key={step.name}>
            <StepLabel
              error={step.hasError}
              StepIconComponent={step.hasError ? StyledCancelIcon : null}
            >
              {step.name}
            </StepLabel>
          </Step>
        ))}
      </Stepper>
    </StyledBox>
  );
};
