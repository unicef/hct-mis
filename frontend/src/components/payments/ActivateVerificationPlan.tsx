import { Box, Button, DialogContent, DialogTitle } from '@material-ui/core';
import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { Dialog } from '../../containers/dialogs/Dialog';
import { DialogActions } from '../../containers/dialogs/DialogActions';
import { DialogContainer } from '../../containers/dialogs/DialogContainer';
import { DialogFooter } from '../../containers/dialogs/DialogFooter';
import { DialogTitleWrapper } from '../../containers/dialogs/DialogTitleWrapper';
import { usePaymentRefetchQueries } from '../../hooks/usePaymentRefetchQueries';
import { useSnackbar } from '../../hooks/useSnackBar';
import { useActivatePaymentVerificationPlanMutation } from '../../__generated__/graphql';

export interface ActivateVerificationPlanProps {
  paymentVerificationPlanId: string;
  cashOrPaymentPlanId: string;
}

export function ActivateVerificationPlan({
  paymentVerificationPlanId,
  cashOrPaymentPlanId,
}: ActivateVerificationPlanProps): React.ReactElement {
  const refetchQueries = usePaymentRefetchQueries(cashOrPaymentPlanId);
  const { t } = useTranslation();
  const [activateDialogOpen, setActivateDialogOpen] = useState(false);

  const { showMessage } = useSnackbar();
  const [mutate] = useActivatePaymentVerificationPlanMutation();
  const activate = async (): Promise<void> => {
    try {
      await mutate({
        variables: { paymentVerificationPlanId },
        refetchQueries,
      });
    } catch (error) {
      /* eslint-disable-next-line no-console */
      console.log('error', error?.graphQLErrors);
      if (
        error?.graphQLErrors?.[0]?.validationErrors
          ?.activateCashPlanPaymentVerification?.phone_numbers
      ) {
        showMessage(
          error?.graphQLErrors?.[0]?.validationErrors?.activateCashPlanPaymentVerification?.phone_numbers.join(
            '\n',
          ),
        );
      } else {
        showMessage(t('Error during activating.'));
      }
    }

    showMessage(t('Verification plan has been activated.'));
  };
  return (
    <>
      <Box p={2}>
        <Button
          color='primary'
          variant='contained'
          onClick={() => setActivateDialogOpen(true)}
          data-cy='button-activate-plan'
        >
          {t('ACTIVATE')}
        </Button>
      </Box>
      <Dialog
        open={activateDialogOpen}
        onClose={() => setActivateDialogOpen(false)}
        scroll='paper'
        aria-labelledby='form-dialog-title'
        maxWidth='md'
      >
        <DialogTitleWrapper>
          <DialogTitle id='scroll-dialog-title'>
            {t('Activate Verification Plan')}
          </DialogTitle>
        </DialogTitleWrapper>
        <DialogContent>
          <DialogContainer>
            <Box p={5}>
              {t(
                'Are you sure you want to activate the Verification Plan for this Cash Plan?',
              )}
            </Box>
          </DialogContainer>
        </DialogContent>
        <DialogFooter>
          <DialogActions>
            <Button onClick={() => setActivateDialogOpen(false)}>CANCEL</Button>
            <Button
              type='submit'
              color='primary'
              variant='contained'
              onClick={() => activate()}
              data-cy='button-submit'
            >
              {t('ACTIVATE')}
            </Button>
          </DialogActions>
        </DialogFooter>
      </Dialog>
    </>
  );
}
