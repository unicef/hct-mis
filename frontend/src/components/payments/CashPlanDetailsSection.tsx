import { Box, Grid, Typography } from '@material-ui/core';
import React from 'react';
import { Doughnut } from 'react-chartjs-2';
import { useTranslation } from 'react-i18next';
import styled from 'styled-components';
import { useBusinessArea } from '../../hooks/useBusinessArea';
import { countPercentage } from '../../utils/utils';
import { CashPlanQuery } from '../../__generated__/graphql';
import { BlackLink } from '../core/BlackLink';
import { LabelizedField } from '../core/LabelizedField';
import { UniversalMoment } from '../core/UniversalMoment';

const Title = styled.div`
  padding-bottom: ${({ theme }) => theme.spacing(8)}px;
`;

const ChartContainer = styled.div`
  width: 150px;
  height: 150px;
`;

const BorderLeftBox = styled.div`
  padding-left: ${({ theme }) => theme.spacing(6)}px;
  border-left: 1px solid #e0e0e0;
  height: 100%;
`;

interface CashPlanDetailsSectionProps {
  cashPlan: CashPlanQuery['cashPlan'];
}

export const CashPlanDetailsSection = ({
  cashPlan,
}: CashPlanDetailsSectionProps): React.ReactElement => {
  const { t } = useTranslation();
  const businessArea = useBusinessArea();

  const bankReconciliationSuccessPercentage = countPercentage(
    cashPlan.bankReconciliationSuccess,
    cashPlan.paymentRecords.totalCount,
  );

  const bankReconciliationErrorPercentage = countPercentage(
    cashPlan.bankReconciliationError,
    cashPlan.paymentRecords.totalCount,
  );

  return (
    <Grid container>
      <Grid item xs={9}>
        <Title>
          <Typography variant='h6'>{t('Cash Plan Details')}</Typography>
        </Title>
        <Grid container>
          {[
            { label: t('PROGRAMME NAME'), value: cashPlan.program.name },
            {
              label: t('PROGRAMME ID'),
              value: (
                <BlackLink
                  to={`/${businessArea}/programs/${cashPlan.program.id}`}
                >
                  {cashPlan.program?.caId}
                </BlackLink>
              ),
            },
            {
              label: t('PAYMENT RECORDS'),
              value: cashPlan.availablePaymentRecordsCount,
            },
            {
              label: t('START DATE'),
              value: <UniversalMoment>{cashPlan.startDate}</UniversalMoment>,
            },
            {
              label: t('END DATE'),
              value: <UniversalMoment>{cashPlan.endDate}</UniversalMoment>,
            },
          ].map((el) => (
            <Grid item xs={4} key={el.label}>
              <Box pt={2} pb={2}>
                <LabelizedField label={el.label}>{el.value}</LabelizedField>
              </Box>
            </Grid>
          ))}
        </Grid>
      </Grid>
      <Grid item xs={3}>
        <BorderLeftBox>
          <Title>
            <Typography variant='h6'>{t('Bank reconciliation')}</Typography>
          </Title>
          <Grid container>
            <Grid item xs={6}>
              <Grid container direction='column'>
                <LabelizedField label={t('SUCCESSFUL')}>
                  <p>{bankReconciliationSuccessPercentage}%</p>
                </LabelizedField>
                <LabelizedField label={t('ERRONEOUS')}>
                  <p>{bankReconciliationErrorPercentage}%</p>
                </LabelizedField>
              </Grid>
            </Grid>
            <Grid item xs={6}>
              <ChartContainer>
                <Doughnut
                  width={200}
                  height={200}
                  options={{
                    maintainAspectRatio: false,
                    cutoutPercentage: 80,
                    legend: {
                      display: false,
                    },
                  }}
                  data={{
                    labels: [t('SUCCESSFUL'), t('ERRONEOUS')],
                    datasets: [
                      {
                        data: [
                          bankReconciliationSuccessPercentage,
                          bankReconciliationErrorPercentage,
                        ],
                        backgroundColor: ['#00509F', '#FFAA1F'],
                        hoverBackgroundColor: ['#00509F', '#FFAA1F'],
                      },
                    ],
                  }}
                />
              </ChartContainer>
            </Grid>
          </Grid>
        </BorderLeftBox>
      </Grid>
    </Grid>
  );
};
