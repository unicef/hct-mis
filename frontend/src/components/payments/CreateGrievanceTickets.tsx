import React, { useState } from 'react';
import { Button, DialogContent, DialogTitle, Box } from '@material-ui/core';
import styled from 'styled-components';
import { Dialog } from '../../containers/dialogs/Dialog';
import { DialogActions } from '../../containers/dialogs/DialogActions';
import { useSnackbar } from '../../hooks/useSnackBar';

export interface Props {
  disabled: boolean;
  grievanceTicketsNumber: number;
}
const DialogTitleWrapper = styled.div`
  border-bottom: 1px solid ${({ theme }) => theme.hctPalette.lighterGray};
`;

const DialogFooter = styled.div`
  padding: 12px 16px;
  margin: 0;
  border-top: 1px solid ${({ theme }) => theme.hctPalette.lighterGray};
  text-align: right;
`;

const DialogContainer = styled.div`
  width: 700px;
`;

export function CreateGrievanceTickets({
  disabled,
  grievanceTicketsNumber,
}: Props): React.ReactElement {
  const [DialogOpen, setDialogOpen] = useState(false);

  const createGrievanceTickets = (): void => {
    setDialogOpen(false);
  };
  return (
    <>
      <Button
        color='primary'
        disabled={disabled}
        onClick={() => setDialogOpen(true)}
        data-cy='button-ed-plan'
      >
        CREATE GRIEVANCE TICKETS
      </Button>
      <Dialog
        open={DialogOpen}
        onClose={() => setDialogOpen(false)}
        scroll='paper'
        aria-labelledby='form-dialog-title'
      >
        <DialogTitleWrapper>
          <DialogTitle id='scroll-dialog-title'>
            Create Grievance Tickets
          </DialogTitle>
        </DialogTitleWrapper>
        <DialogContent>
          <DialogContainer>
            <Box p={5}>
              <div>
                You will generate {grievanceTicketsNumber} grievance ticket
                {grievanceTicketsNumber === 1 ? null : 's'}. Are you sure you
                want to continue?
              </div>
            </Box>
          </DialogContainer>
        </DialogContent>
        <DialogFooter>
          <DialogActions>
            <Button onClick={() => setDialogOpen(false)}>CANCEL</Button>
            <Button
              type='submit'
              color='primary'
              variant='contained'
              onClick={() => createGrievanceTickets()}
              data-cy='button-submit'
            >
              Continue
            </Button>
          </DialogActions>
        </DialogFooter>
      </Dialog>
    </>
  );
}
