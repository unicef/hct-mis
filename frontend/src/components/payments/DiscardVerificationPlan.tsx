import React, { useState } from 'react';
import { Button, DialogContent, DialogTitle, Box } from '@material-ui/core';
import CheckRoundedIcon from '@material-ui/icons/CheckRounded';
import styled from 'styled-components';
import ClearIcon from '@material-ui/icons/Clear';
import { Dialog } from '../../containers/dialogs/Dialog';
import { DialogActions } from '../../containers/dialogs/DialogActions';

export function DiscardVerificationPlan(): React.ReactElement {
  const [finishDialogOpen, setFinishDialogOpen] = useState(false);

  // const submit = async (): Promise<void> => {
  //   // const { errors } = await mutate();
  //   const errors = [];
  //   if (errors) {
  //     showMessage('Error while submitting');
  //     return;
  //   }
  //   showMessage('New verification plan created.');
  // };
  const DialogTitleWrapper = styled.div`
    border-bottom: 1px solid ${({ theme }) => theme.hctPalette.lighterGray};
  `;

  const DialogFooter = styled.div`
    padding: 12px 16px;
    margin: 0;
    border-top: 1px solid ${({ theme }) => theme.hctPalette.lighterGray};
    text-align: right;
  `;

  const DialogContainer = styled.div`
    width: 700px;
  `;

  return (
    <>
      <Box p={2}>
        <Button
          color='secondary'
          startIcon={<ClearIcon />}
          onClick={() => setFinishDialogOpen(true)}
          data-cy='button-discard-plan'
        >
          DISCARD
        </Button>
      </Box>
      <Dialog
        open={finishDialogOpen}
        onClose={() => setFinishDialogOpen(false)}
        scroll='paper'
        aria-labelledby='form-dialog-title'
      >
        <DialogTitleWrapper>
          <DialogTitle id='scroll-dialog-title'>
            Discard Verification Plan
          </DialogTitle>
        </DialogTitleWrapper>
        <DialogContent>
          <DialogContainer>
            <Box p={5}>
              <div>
                Payment verification for households who have responded will not
                be visible
              </div>
              <div>or valid anymore. Are you sure?</div>
            </Box>
          </DialogContainer>
        </DialogContent>
        <DialogFooter>
          <DialogActions>
            <Button onClick={() => setFinishDialogOpen(false)}>CANCEL</Button>
            <Button
              type='submit'
              color='secondary'
              variant='contained'
              onClick={() => console.log('discard')}
              data-cy='button-submit'
            >
              DISCARD
            </Button>
          </DialogActions>
        </DialogFooter>
      </Dialog>
    </>
  );
}
