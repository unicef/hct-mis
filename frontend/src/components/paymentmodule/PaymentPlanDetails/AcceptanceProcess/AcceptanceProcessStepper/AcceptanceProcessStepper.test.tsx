import { act } from '@testing-library/react';
import React from 'react';
import wait from 'waait';
import { fakeApolloPaymentPlan } from '../../../../../../fixtures/payments/fakeApolloPaymentPlan';
import { render } from '../../../../../testUtils/testUtils';
import { AcceptanceProcessStepper } from './AcceptanceProcessStepper';

describe('components/paymentmodule/PaymentPlanDetails/AcceptanceProcess/AcceptanceProcessStepper', () => {
  it('should render default step Sent for Approval Date', async () => {
    const { container } = render(
      <AcceptanceProcessStepper
        acceptanceProcess={fakeApolloPaymentPlan.approvalProcess.edges[0].node}
        approvalNumberRequired={1}
        authorizationNumberRequired={1}
        financeReviewNumberRequired={1}
      />,
    );
    await act(() => wait(0)); // wait for response
    expect(container).toMatchSnapshot();
  });
  it('should render step Sent for Approval Date', async () => {
    const { container } = render(
      <AcceptanceProcessStepper
        acceptanceProcess={{...fakeApolloPaymentPlan.approvalProcess.edges[0].node, sentForApprovalDate: '2020-01-01'}}
        approvalNumberRequired={1}
        authorizationNumberRequired={1}
        financeReviewNumberRequired={1}
      />,
    );
    await act(() => wait(0)); // wait for response
    expect(container).toMatchSnapshot();
  });
  it('should render step Sent for Authorization Date', async () => {
    const { container } = render(
      <AcceptanceProcessStepper
        acceptanceProcess={{...fakeApolloPaymentPlan.approvalProcess.edges[0].node, sentForAuthorizationDate: '2020-01-01'}}
        approvalNumberRequired={1}
        authorizationNumberRequired={1}
        financeReviewNumberRequired={1}
      />,
    );
    await act(() => wait(0)); // wait for response
    expect(container).toMatchSnapshot();
  });
  it('should render step Sent for Finance Review Date', async () => {
    const { container } = render(
      <AcceptanceProcessStepper
        acceptanceProcess={{...fakeApolloPaymentPlan.approvalProcess.edges[0].node, sentForFinanceReviewDate: '2020-01-02'}}
        approvalNumberRequired={1}
        authorizationNumberRequired={1}
        financeReviewNumberRequired={1}
      />,
    );
    await act(() => wait(0)); // wait for response
    expect(container).toMatchSnapshot();
  });
});
