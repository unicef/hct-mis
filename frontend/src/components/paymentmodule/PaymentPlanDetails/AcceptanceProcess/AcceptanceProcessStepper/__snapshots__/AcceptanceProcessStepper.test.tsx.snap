// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/paymentmodule/PaymentPlanDetails/AcceptanceProcess/AcceptanceProcessStepper should render default step Sent for Approval Date 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-1 sc-bdVaJa UoSJW"
  >
    <div
      class="MuiPaper-root MuiStepper-root MuiStepper-horizontal MuiPaper-elevation0"
    >
      <div
        class="MuiStep-root MuiStep-horizontal MuiStep-completed"
      >
        <span
          class="MuiStepLabel-root MuiStepLabel-horizontal"
        >
          <span
            class="MuiStepLabel-iconContainer"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-active MuiStepIcon-completed"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 0a12 12 0 1 0 0 24 12 12 0 0 0 0-24zm-2 17l-5-5 1.4-1.4 3.6 3.6 7.6-7.6L19 8l-9 9z"
              />
            </svg>
          </span>
          <span
            class="MuiStepLabel-labelContainer"
          >
            <span
              class="MuiTypography-root MuiStepLabel-label MuiStepLabel-completed MuiStepLabel-active MuiTypography-body2 MuiTypography-displayBlock"
            >
              Approval (1/1)
            </span>
          </span>
        </span>
      </div>
      <div
        class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled"
      >
        <span
          class="MuiStepConnector-line MuiStepConnector-lineHorizontal"
        />
      </div>
      <div
        class="MuiStep-root MuiStep-horizontal"
      >
        <span
          class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled"
        >
          <span
            class="MuiStepLabel-iconContainer"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiStepIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <circle
                cx="12"
                cy="12"
                r="12"
              />
              <text
                class="MuiStepIcon-text"
                text-anchor="middle"
                x="12"
                y="16"
              >
                2
              </text>
            </svg>
          </span>
          <span
            class="MuiStepLabel-labelContainer"
          >
            <span
              class="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
            >
              Authorization (0/1)
            </span>
          </span>
        </span>
      </div>
      <div
        class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled"
      >
        <span
          class="MuiStepConnector-line MuiStepConnector-lineHorizontal"
        />
      </div>
      <div
        class="MuiStep-root MuiStep-horizontal"
      >
        <span
          class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled"
        >
          <span
            class="MuiStepLabel-iconContainer"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiStepIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <circle
                cx="12"
                cy="12"
                r="12"
              />
              <text
                class="MuiStepIcon-text"
                text-anchor="middle"
                x="12"
                y="16"
              >
                3
              </text>
            </svg>
          </span>
          <span
            class="MuiStepLabel-labelContainer"
          >
            <span
              class="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
            >
              Finance Review (0/1)
            </span>
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`components/paymentmodule/PaymentPlanDetails/AcceptanceProcess/AcceptanceProcessStepper should render step Sent for Approval Date 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-2 sc-bdVaJa UoSJW"
  >
    <div
      class="MuiPaper-root MuiStepper-root MuiStepper-horizontal MuiPaper-elevation0"
    >
      <div
        class="MuiStep-root MuiStep-horizontal MuiStep-completed"
      >
        <span
          class="MuiStepLabel-root MuiStepLabel-horizontal"
        >
          <span
            class="MuiStepLabel-iconContainer"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-active MuiStepIcon-completed"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 0a12 12 0 1 0 0 24 12 12 0 0 0 0-24zm-2 17l-5-5 1.4-1.4 3.6 3.6 7.6-7.6L19 8l-9 9z"
              />
            </svg>
          </span>
          <span
            class="MuiStepLabel-labelContainer"
          >
            <span
              class="MuiTypography-root MuiStepLabel-label MuiStepLabel-completed MuiStepLabel-active MuiTypography-body2 MuiTypography-displayBlock"
            >
              Approval (1/1)
            </span>
          </span>
        </span>
      </div>
      <div
        class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled"
      >
        <span
          class="MuiStepConnector-line MuiStepConnector-lineHorizontal"
        />
      </div>
      <div
        class="MuiStep-root MuiStep-horizontal"
      >
        <span
          class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled"
        >
          <span
            class="MuiStepLabel-iconContainer"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiStepIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <circle
                cx="12"
                cy="12"
                r="12"
              />
              <text
                class="MuiStepIcon-text"
                text-anchor="middle"
                x="12"
                y="16"
              >
                2
              </text>
            </svg>
          </span>
          <span
            class="MuiStepLabel-labelContainer"
          >
            <span
              class="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
            >
              Authorization (0/1)
            </span>
          </span>
        </span>
      </div>
      <div
        class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled"
      >
        <span
          class="MuiStepConnector-line MuiStepConnector-lineHorizontal"
        />
      </div>
      <div
        class="MuiStep-root MuiStep-horizontal"
      >
        <span
          class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled"
        >
          <span
            class="MuiStepLabel-iconContainer"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiStepIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <circle
                cx="12"
                cy="12"
                r="12"
              />
              <text
                class="MuiStepIcon-text"
                text-anchor="middle"
                x="12"
                y="16"
              >
                3
              </text>
            </svg>
          </span>
          <span
            class="MuiStepLabel-labelContainer"
          >
            <span
              class="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
            >
              Finance Review (0/1)
            </span>
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`components/paymentmodule/PaymentPlanDetails/AcceptanceProcess/AcceptanceProcessStepper should render step Sent for Authorization Date 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-3 sc-bdVaJa UoSJW"
  >
    <div
      class="MuiPaper-root MuiStepper-root MuiStepper-horizontal MuiPaper-elevation0"
    >
      <div
        class="MuiStep-root MuiStep-horizontal MuiStep-completed"
      >
        <span
          class="MuiStepLabel-root MuiStepLabel-horizontal"
        >
          <span
            class="MuiStepLabel-iconContainer"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-completed"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 0a12 12 0 1 0 0 24 12 12 0 0 0 0-24zm-2 17l-5-5 1.4-1.4 3.6 3.6 7.6-7.6L19 8l-9 9z"
              />
            </svg>
          </span>
          <span
            class="MuiStepLabel-labelContainer"
          >
            <span
              class="MuiTypography-root MuiStepLabel-label MuiStepLabel-completed MuiTypography-body2 MuiTypography-displayBlock"
            >
              Approval (1/1)
            </span>
          </span>
        </span>
      </div>
      <div
        class="MuiStepConnector-root MuiStepConnector-horizontal MuiStepConnector-active"
      >
        <span
          class="MuiStepConnector-line MuiStepConnector-lineHorizontal"
        />
      </div>
      <div
        class="MuiStep-root MuiStep-horizontal"
      >
        <span
          class="MuiStepLabel-root MuiStepLabel-horizontal"
        >
          <span
            class="MuiStepLabel-iconContainer"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-active"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <circle
                cx="12"
                cy="12"
                r="12"
              />
              <text
                class="MuiStepIcon-text"
                text-anchor="middle"
                x="12"
                y="16"
              >
                2
              </text>
            </svg>
          </span>
          <span
            class="MuiStepLabel-labelContainer"
          >
            <span
              class="MuiTypography-root MuiStepLabel-label MuiStepLabel-active MuiTypography-body2 MuiTypography-displayBlock"
            >
              Authorization (0/1)
            </span>
          </span>
        </span>
      </div>
      <div
        class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled"
      >
        <span
          class="MuiStepConnector-line MuiStepConnector-lineHorizontal"
        />
      </div>
      <div
        class="MuiStep-root MuiStep-horizontal"
      >
        <span
          class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled"
        >
          <span
            class="MuiStepLabel-iconContainer"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiStepIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <circle
                cx="12"
                cy="12"
                r="12"
              />
              <text
                class="MuiStepIcon-text"
                text-anchor="middle"
                x="12"
                y="16"
              >
                3
              </text>
            </svg>
          </span>
          <span
            class="MuiStepLabel-labelContainer"
          >
            <span
              class="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
            >
              Finance Review (0/1)
            </span>
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`components/paymentmodule/PaymentPlanDetails/AcceptanceProcess/AcceptanceProcessStepper should render step Sent for Finance Review Date 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-4 sc-bdVaJa UoSJW"
  >
    <div
      class="MuiPaper-root MuiStepper-root MuiStepper-horizontal MuiPaper-elevation0"
    >
      <div
        class="MuiStep-root MuiStep-horizontal MuiStep-completed"
      >
        <span
          class="MuiStepLabel-root MuiStepLabel-horizontal"
        >
          <span
            class="MuiStepLabel-iconContainer"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-completed"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 0a12 12 0 1 0 0 24 12 12 0 0 0 0-24zm-2 17l-5-5 1.4-1.4 3.6 3.6 7.6-7.6L19 8l-9 9z"
              />
            </svg>
          </span>
          <span
            class="MuiStepLabel-labelContainer"
          >
            <span
              class="MuiTypography-root MuiStepLabel-label MuiStepLabel-completed MuiTypography-body2 MuiTypography-displayBlock"
            >
              Approval (1/1)
            </span>
          </span>
        </span>
      </div>
      <div
        class="MuiStepConnector-root MuiStepConnector-horizontal"
      >
        <span
          class="MuiStepConnector-line MuiStepConnector-lineHorizontal"
        />
      </div>
      <div
        class="MuiStep-root MuiStep-horizontal"
      >
        <span
          class="MuiStepLabel-root MuiStepLabel-horizontal"
        >
          <span
            class="MuiStepLabel-iconContainer"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiStepIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <circle
                cx="12"
                cy="12"
                r="12"
              />
              <text
                class="MuiStepIcon-text"
                text-anchor="middle"
                x="12"
                y="16"
              >
                2
              </text>
            </svg>
          </span>
          <span
            class="MuiStepLabel-labelContainer"
          >
            <span
              class="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
            >
              Authorization (0/1)
            </span>
          </span>
        </span>
      </div>
      <div
        class="MuiStepConnector-root MuiStepConnector-horizontal MuiStepConnector-active"
      >
        <span
          class="MuiStepConnector-line MuiStepConnector-lineHorizontal"
        />
      </div>
      <div
        class="MuiStep-root MuiStep-horizontal"
      >
        <span
          class="MuiStepLabel-root MuiStepLabel-horizontal"
        >
          <span
            class="MuiStepLabel-iconContainer"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-active"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <circle
                cx="12"
                cy="12"
                r="12"
              />
              <text
                class="MuiStepIcon-text"
                text-anchor="middle"
                x="12"
                y="16"
              >
                3
              </text>
            </svg>
          </span>
          <span
            class="MuiStepLabel-labelContainer"
          >
            <span
              class="MuiTypography-root MuiStepLabel-label MuiStepLabel-active MuiTypography-body2 MuiTypography-displayBlock"
            >
              Finance Review (0/1)
            </span>
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;
